### ISSUE TITLE
Function composition / flow operator

### ISSUE THREAD

**Issue by kabo** [DanielXMoore/Civet#423]
I use ramda a lot and I was curious if civet could support a few things natively. Something I do a lot with ramda is create a function using pipe. For example:

```typescript
const doTheThing = R.pipe(
  R.pluck('myprop'),
  R.map(R.toUpper)
)

console.log(doTheThing([{myprop: 'hello'}])) // outputs [HELLO]
```

My understanding is that the pipe operator in civet currently requires a start value, so one can't easily compose functions using pipe. You can see my attempts at this in https://github.com/DanielXMoore/Civet/issues/75#issuecomment-1448930717 

It would be really neat if there was some sort of function composition / flow operator (see [pipe vs flow in fp-ts](https://rlee.dev/practical-guide-to-fp-ts-part-1) and [F# Pipelining vs. Composition](https://spin.atomicobject.com/2021/08/26/fsharp-operators-pipeline-composition/) for good explanations of these two concepts).

F# uses `>>` for composition. So these two would be equivalent:

```typescript
input := [myprop: 'hello']

// using pipe, |>
input
  |> .map .myprop
  |> .map .toUpperCase() 
  |> console.log

// using flow, >>
doTheThing := .map .myprop >> .map .toUpperCase()
console.log doTheThing input

**Comment by orenelbaum** [comment]
wouldn't you need parens?
`doTheThing := (.map .myprop) >> (.map .toUpperCase())`

**Comment by kabo** [comment]
> wouldn't you need parens? `doTheThing := (.map .myprop) >> (.map .toUpperCase())`

Possibly. I learned about the existence of civet yesterday, please excuse any mistakes in my civet code :slightly_smiling_face: 

I was thinking since the pipe code works fine without parens it'd be nice if this worked the same. i.e. to convert from pipe to flow:
1. remove `input |>`
2. replace all `|>` with `>>`
3. there is no step 3

**Comment by shurko0x4cfd** [comment]
Just for aesthetic reasons, I prefer >> instead of |>.

**Comment by edemaine** [comment]
I don't think we can use `>>` (until we have operator overloading) because `f >> g` already has a meaning in JavaScript (shift right).

Oren mentioned on Discord that these examples could be reasonably represented in Civet as `& |> f |> g`, which would mean `$ => g(f($))`. This doesn't work yet but would be a natural extension to `&` notation (function shorthand).

Do you think that would suffice, or is this too verbose?

**Comment by shurko0x4cfd** [comment]
Operator overload looks exciting and desirable.

**Comment by kabo** [comment]
> Oren mentioned on Discord that these examples could be reasonably represented in Civet as `& |> f |> g`, which would mean `$ => g(f($))`. This doesn't work yet but would be a natural extension to `&` notation (function shorthand).
> 
> Do you think that would suffice, or is this too verbose?

Yes, I forgot to mention I did try that, it felt natural, so personally I think that's a good solution.

**Comment by STRd6** [comment]
Released in v0.5.81
